File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/writing_data/sample0.py
import dv_processing as dv

config = dv.io.MonoCameraWriter.Config("DVXplorer_sample")

# Define VGA resolution for this camera
resolution = (640, 480)

# Add an event stream with a resolution
config.addEventStream(resolution)

# Add frame stream with a resolution
config.addFrameStream(resolution)

# Add IMU stream
config.addImuStream()

# Add trigger stream
config.addTriggerStream()

# Create the writer instance with the configuration structure
writer = dv.io.MonoCameraWriter("mono_writer_sample.aedat4", config)

print(f"Is event stream available? {str(writer.isEventStreamConfigured())}")
print(f"Is frame stream available? {str(writer.isFrameStreamConfigured())}")
print(f"Is imu stream available? {str(writer.isImuStreamConfigured())}")
print(f"Is trigger stream available? {str(writer.isTriggerStreamConfigured())}")

File: source_code/ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/writing_data/sample0.cpp
#include <dv-processing/io/mono_camera_writer.hpp>

int main() {
	dv::io::MonoCameraWriter::Config config("DVXplorer_sample");

	// Sample VGA resolution, same as the DVXplorer camera
	const cv::Size resolution(640, 480);

	// Add an event stream with a resolution
	config.addEventStream(resolution);

	// Add frame stream with a resolution
	config.addFrameStream(resolution);

	// Add IMU stream
	config.addImuStream();

	// Add trigger stream
	config.addTriggerStream();

	// Create the writer instance with the configuration structure
	dv::io::MonoCameraWriter writer("mono_writer_sample.aedat4", config);

	// Print which streams were configured
	std::cout << std::boolalpha;
	std::cout << "Is event stream available? " << writer.isEventStreamConfigured() << std::endl;
	std::cout << "Is frame stream available? " << writer.isFrameStreamConfigured() << std::endl;
	std::cout << "Is imu stream available? " << writer.isImuStreamConfigured() << std::endl;
	std::cout << "Is trigger stream available? " << writer.isTriggerStreamConfigured() << std::endl;

	return 0;
}
